client/components/head/index.jsx

```diff
@@ -71,12 +78,34 @@ const Head = ( { title = 'WordPress.com', faviconURL, children, cdn, branchName
 					href={ '/calypso/manifest.json?branch=' + encodeURIComponent( branchName ) }
 				/>
 			) }
-
 			<link
-				rel="stylesheet"
+				rel="preload"
 				href="https://fonts.googleapis.com/css?family=Noto+Serif:400,400i,700,700i&subset=cyrillic,cyrillic-ext,greek,greek-ext,latin-ext,vietnamese"
+				as="style"
 			/>
-
+			<noscript>
+				<link
+					rel="stylesheet"
+					href="https://fonts.googleapis.com/css?family=Noto+Serif:400,400i,700,700i&subset=cyrillic,cyrillic-ext,greek,greek-ext,latin-ext,vietnamese"
+				/>
+			</noscript>
+			{ /* eslint-disable react/no-danger */ }
+			<script
```

Comment:
@sgomes I tried that, but React seems to hate the `onLoad` attribute on a link tag. I couldn't get it to come through the SSR / client rehydrate bits successfully. So I fell back to this...

If you can make it work, I'd love to use the simpler style.

Reply:
This part of the markup is never rehydrated nor rendered by React on client. On client we render React UI into the `<div id="wpcom">` root element and everything around it is treated as static HTML. Yes, it's been rendered as JSX markup on the server and then serialized into a string, but that's all. Server-side event listeners are noops.

It used to be a PUG template some time ago.

Reply:
Hmmm... so is there a way to get a literal onload handler to flow out through SSR?

Reply:
The idea behind using React was to reuse components. If that's, however, something which gets in the way I'm all in just using e.g. template strings.

Reply:
I personally like having a high level of control over the app shell, without any frameworks in between my code and the end user's browser. I think the compromise of higher maintenance costs is acceptable as a tradeoff for the ability to fine-tune loading performance for end-users.

That may or may not be the right choice for Calypso. I suspect it is, but my opinion is clearly biased and I'm still quite new to the project :)

Reply:
The app shell doesn't have any React UI components: it's mostly `script` and `link` elements for loading assets, inline scripts for initial state and boot code etc. React is not very helpful here IMO.

One of the opportunities that React gives us here is streaming the response. When streaming, the browser can already be downloading the CSS stylesheets at a time when the server is still SSR-ing the initial markup. Let's not make future streaming harder if we decide for a simpler non-React approach.

Reply:
Note that e.g. the 404 template uses the `EmptyContent` component:

https://github.com/Automattic/wp-calypso/blob/3f6213e0606382b88e2ef399028c1b1354c3f538/client/document/404.jsx#L27-L33

It was probably me who opted to use JSX for the app shell. I was hoping to be able to re-use components, and thus maybe also lower the bar a bit for more SSR'd sections.

React at this level might also help us one day adopt [`renderToNodeStream()`](https://reactjs.org/docs/react-dom-server.html#rendertonodestream), possibly allowing us to SSR  logged-in parts of Calypso more easily (since we might not need to cache as much to take off load from node's event loop in order not to block it)?

_This SSR Promo brought to you by @ockham._